libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -395,8 +395,7 @@ void NavEKF3_core::InitialiseVariables()
     memset(&yawAngDataDelayed, 0, sizeof(yawAngDataDelayed));
 
     // external nav data fusion
-    memset(&extNavDataNew, 0, sizeof(extNavDataNew));
-    memset(&extNavDataDelayed, 0, sizeof(extNavDataDelayed));
+    extNavDataDelayed.time_ms = 0;
```

Comment:
why don't you need to zero all the extNavDataDelayed data like it was done previously ? 
In case of reset we need the value to be cleared, right ?
I don't have check deeply the EKF flow, but it seems that some functions are using extNavDataDelayed data without checking first extNavDataDelayed.time_ms

a simple extNavDataDelayed = {}; whould keep the current zeroing (even beeing converted to memset by the compiler) and remove the warning

Reply:
ok, I've changed it to that, although there is a bit of flash cost (compiler needs to create a constructor for the type)